### PR TITLE
[GPU] Implement primitive_node method to check runtime skippable

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *Implement primitive_node method to check runtime skippable*

### Tickets:
 - *137650*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@ void mark_runtime_skippable_nodes::run(program& p) {
         // Set gathers that might be skipped at runtime as can_be_optimized.
         // If not set, memory dependency will not work for the nodes that are skipped at runtime
         program_helpers::do_for_types<gather>(*node, [](gather_node& node){
+            node.is_runtime_skippable(true);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp:23]: We should set this only when the node is set as can_be_optimized finally. 
So please move this to line 40. Likewise for other nodes.

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp]: Done.

File: src/plugins/intel_gpu/src/graph/include/program_node.h [hunk 2]
```diff
@@ -301,6 +301,10 @@ struct program_node {
     bool can_be_optimized() const { return optimized; }
     void can_be_optimized(bool opt) { optimized = opt; }
 
+    // check/set if the node is runtime skippable
+    bool is_runtime_skippable() const { return runtime_skippable; }
+    void is_runtime_skippable(bool skippable) { runtime_skippable = skippable; }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/include/program_node.h:306]: Minor change, but how about change this to "set_runtime_skippable" to reduce confusion of the purpose of the function?

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/include/program_node.h]: Done

**Review by yeonbok — APPROVED** [review]

**Pull request merged**